tracee-ebpf/tracee/events_processor.go

```diff
@@ -155,19 +156,19 @@ func (t *Tracee) processEvent(ctx *context, args map[string]interface{}, argMeta
 
 					// create an in-memory profile
 					if t.config.Capture.Profile {
-						t.updateProfile(fmt.Sprintf("%s:%d", filepath.Join(destinationDirPath, fmt.Sprintf("exec.%s", filepath.Base(filePath))), sourceFileCtime), ctx.Ts)
+						t.updateProfile(fmt.Sprintf("%s:%d", filepath.Join(destinationDirPath, fmt.Sprintf("exec.%s", filepath.Base(filePath))), castedSourceFileCtime), ctx.Ts)
 					}
 
 					//don't capture same file twice unless it was modified
 					lastCtime, ok := t.capturedFiles[capturedFileID]
-					if !ok || lastCtime != sourceFileCtime {
+					if !ok || lastCtime != castedSourceFileCtime {
 						//capture
 						err = CopyFileByPath(sourceFilePath, destinationFilePath)
 						if err != nil {
 							return err
 						}
 						//mark this file as captured
-						t.capturedFiles[capturedFileID] = sourceFileCtime
+						t.capturedFiles[capturedFileID] = castedSourceFileCtime
```

Comment:
I think this is the only place that requires to cast the ctime, right?
If so, you can avoid the declaration and usage of castedSourceFileCtime in other places, and only cast here

Reply:
we use it also in line 164 - need to have the casted variable, no?

Reply:
You can compare without a cast, knowing that timestamp will never be longer than 63 bits (thus you can compare the "signed" timestamp to unsigned)

Reply:
not sure you can actually "Invalid operation: lastCtime != sourceFileCtime (mismatched types int64 and uint64)"